pkg/server/sessions/manager.go

```diff
@@ -0,0 +1,244 @@
+/*
+Copyright 2021 CodeNotary, Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sessions
+
+import (
+	"context"
+	"github.com/codenotary/immudb/pkg/auth"
+	"github.com/codenotary/immudb/pkg/errors"
+	"github.com/codenotary/immudb/pkg/logger"
+	"github.com/codenotary/immudb/pkg/server/sessions/internal/transactions"
+	"github.com/rs/xid"
+	"os"
+	"sync"
+	"time"
+)
+
+var ErrGuardAlreadyRunning = errors.New("session guard already launched")
+var ErrGuardNotRunning = errors.New("session guard not running")
+
+var guard *manager
```

Comment:
no global vars....

Reply:
aaargh ... is a leftover of a singleton.